tachyon/math/polynomial/domains/general_evaluation_domain_unittest.cc

```diff
@@ -0,0 +1,103 @@
+// Copyright 2022 arkworks contributors
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.arkworks and the LICENCE-APACHE.arkworks
+// file.
+
+#include "tachyon/math/polynomial/domains/general_evaluation_domain.h"
+
+#include "gtest/gtest.h"
+
+#include "tachyon/math/elliptic_curves/bls/bls12_381/fr.h"
+#include "tachyon/math/elliptic_curves/bls/bls12_381/g1.h"
+#include "tachyon/math/elliptic_curves/bn/bn384_small_two_adicity/fr.h"
+#include "tachyon/math/polynomial/polynomials/univariate/univariate_polynomial.h"
+
+namespace tachyon::math {
+
+namespace {
+
+class GeneralEvaluationDomainTest : public testing::Test {
+ public:
+  static void SetUpTestSuite() {
+    bls12_381::Fr::Init();
+    bn384_small_two_adicity::Fr::Init();
+  }
+
+  GeneralEvaluationDomainTest() = default;
+  GeneralEvaluationDomainTest(const GeneralEvaluationDomainTest&) = delete;
+  GeneralEvaluationDomainTest& operator=(const GeneralEvaluationDomainTest&) =
+      delete;
+  ~GeneralEvaluationDomainTest() override = default;
+};
+
+}  // namespace
+
+TEST_F(GeneralEvaluationDomainTest, VanishingPolynomialEvaluation) {
+  for (size_t coeffs = 1; coeffs < 10; ++coeffs) {
+    GeneralEvaluationDomain<bls12_381::Fr> domain(coeffs);
+    auto z = domain.GetVanishingPolynomial();
+    for (size_t _i = 0; _i < 100; ++_i) {
```

Comment:
```suggestion
    for (size_t i = 0; i < 100; ++i) {
```

Reply:
By the way, do we need to loop this test?

Reply:
> By the way, do we need to loop this test?

I think it depends on the point of view. I'm not sure but in the test code of arkworks' they have a loop in this test.